src/Validation/Validator.php

```diff
@@ -2339,7 +2339,8 @@ protected function isEmpty($data, $flags)
             return true;
         }
 
-        if ($data === [] && ($flags & self::EMPTY_ARRAY)) {
+        $arrayTypes = (self::EMPTY_ARRAY | self::EMPTY_DATE | self::EMPTY_TIME | self::EMPTY_FILE);
```

Comment:
The braces are not necessary. 4.x ruleset will complain about it when merging :).